### PR TITLE
[RPS-120] Tune search

### PR DESCRIPTION
**Pull request by wojtek-oledzki**
Main focus of this commit was to configure weights for selected
NodeTypes and properties.

As a side effect, also these things are part of it:
* Switch to fluent search
* Add lorme ipsum test content
* Introduce "common" namespace/configuration/package
* More acceptance tests hardening

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/hst/configurations/common/sitemap.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+definitions:
+  config:
+    /hst:hst/hst:configurations/common/hst:sitemap:
+      jcr:primaryType: hst:sitemap
+      /root:
+        jcr:primaryType: hst:sitemapitem
+        hst:componentconfigurationid: hst:pages/home
+        hst:relativecontentpath: publications/test-document
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-config/hst/configurations/common/sitemap.yaml]: this does not exist anymore :) but I guess we don't care to much as we don't have homepage anyway :D

File: docs/what-if.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# What If?
```
Comment (ziemsky) [inline docs/what-if.md]: How about adding a reference to this file from the main README.MD?

Comment (wojtek-oledzki) [reply docs/what-if.md]: d'oh 

D'oh

**Comment by woledzki** [comment]
corporatewebsite.zip file removed :D @ziemsky

**Comment by woledzki** [comment]
I will update README tomorrow

**Comment by wojtek-oledzki** [comment]
README updated @ziemsky 

ready for re-review :)

File: docs/what-if/boost-document-property.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+At this point I still don't know how to get score (boost) set on individual fields
+to be used when searching entire NodeScope.
+
+That's why in your search component you have to explicitly search for the query in
+the selected fields and NodeScope as well
+
+* `publicationsystem:Title`
+* `publicationsystem:Summar`
```
Comment (ngta1) [inline docs/what-if/boost-document-property.md]: small typo...

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/search-main.ftl [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        </article>
+      </#list>
+      <#if cparam.showPagination>
+        <#include "../include/pagination.ftl">
+      </#if>
+    </div>
+  </#if>
+<#else>
+  <h3>Please fill in a search term.</h3>
+</#if>
```
Comment (ngta1) [inline repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/search-main.ftl]: This file is not used for search results, so could be removed. The search results page is located at freemarker/publication/searchresults.ftl

Comment (woledzki) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/search-main.ftl]: I'm on it...

File: repository-data/application/src/main/resources/hcm-config/hst/configurations/common/sitemap.yaml (hunk 1 above)
Comment (ziemsky) [reply repository-data/application/src/main/resources/hcm-config/hst/configurations/common/sitemap.yaml]: Indeed, no value in doing anything about this until we have specific requirements for the homepage.

**Pull request merged**